### PR TITLE
Update shadow commands for relay test usages

### PR DESCRIPTION
**Pull request by clearloop**
## Steps

+ [x] Fix the duplicated epoch
+ [x] Support multiple formats for `shadow proof`
+ [x] add mmr_proof in api

### CODE REVIEW

**Comment by clearloop** [comment]
## RESTFUL API

+ Proposal

```
POST /api/v1/proposal

[
  [
    {
      "eth_header": {
        "author": "string",
        "difficulty": "string",
        "extra_data": "string",
        "gas_limit": "string",
        "gas_used": "string",
        "hash": "string",
        "log_bloom": "string",
        "number": "string",
        "parent_hash": "string",
        "receipts_root": "string",
        "seal": [
          "string"
        ],
        "state_root": "string",
        "timestamp": "string",
        "transactions_root": "string",
        "uncles_hash": "string"
      },
      "mmr_proof": [
        "string"
      ],
      "proof": [
        {
          "dag_nodes": [
            "string"
          ],
          "proof": [
            "string"
          ]
        }
      ],
      "root": "string"
    }
  ]
]
Headers:
```

+ Receipt

```
GET /api/v1/receipt/:tx

[
  {
    "mmr_proof": [
      "string"
    ],
    "receipt_proof": {
      "header_hash": "string",
      "index": "string",
      "proof": "string"
    }
  }
]
```

**Comment by clearloop** [comment]
There are some compilation bugs in `linux` system, will be fixed in next pr.

**Comment by HackFisher** [comment]
mmr in proposal should not be called as mmr_proof, it is a hash represeting root of mmr. Just using mmr_root.

And what is proof? Is it ethash_proof?

**Comment by AurevoirXavier** [comment]
> mmr in proposal should not be called as mmr_proof, it is a hash represeting root of mmr. Just using mmr_root.
> 
> And what is proof? Is it ethash_proof?

I think the root should be rename to mmr_root. And the mmr_proof looks good.

**Comment by clearloop** [comment]
> mmr in proposal should not be called as mmr_proof, it is a hash represeting root of mmr. Just using mmr_root.
> 
> And what is proof? Is it ethash_proof?

+ A hash represeting root of mmr 

`root`

> ethashproof

`proof`, this field adapts the old API.

> mmr proof by the pos of block number

`mmr_proof`

File: api/docs/docs.go [hunk 1]
```diff
@@ -32,7 +32,380 @@ var doc = `{
...
+            }
+        },
+        "core.GetEthHeaderWithProofJSONResp": {
+            "type": "object",
+            "properties": {
+                "eth_header": {
+                    "type": "object",
+                    "$ref": "#/definitions/eth.DarwiniaEthHeaderHexFormat"
+                },
+                "mmr_proof": {
```
Comment (HackFisher) [inline api/docs/docs.go:239]: use mmr_root

File: api/docs/docs.go [hunk 2]
```diff
@@ -32,7 +32,380 @@ var doc = `{
...
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    }
+                },
+                "proof": {
```
Comment (HackFisher) [inline api/docs/docs.go:245]: use ethash_proof

**Comment by clearloop** [comment]
> fix proposal api para names

## For the first API

+ The `root` field can be renamed to `mmr_root`
+ The old version of `dj` will be broken after renaming `proof` to `ethash_proof`.
+ The `mmr_proof` might not necessary for this API

**Comment by HackFisher** [comment]
> > mmr in proposal should not be called as mmr_proof, it is a hash represeting root of mmr. Just using mmr_root.
> 
> > 
> 
> > And what is proof? Is it ethash_proof?
> 
> 
> 
> I think the root should be rename to mmr_root. And the mmr_proof looks good.
> 
> 
> 
> Yes, proof should be ethash_proof here.

Please take a look at required changes, What I was refering is proposal api, not the on in check receipt api.

File: api/docs/docs.go (hunk 1 above)
Comment (clearloop) [reply api/docs/docs.go:239]: This is actually mmr_proof, `[]string`, the `mmr_root` is `root` in this pr, this field is just for test for now.

File: api/docs/docs.go (hunk 2 above)
Comment (clearloop) [reply api/docs/docs.go:245]: `dj` will be broken after changing this.

File: api/docs/docs.go (hunk 1 above)
Comment (AurevoirXavier) [reply api/docs/docs.go:239]: I think this field is ok.

File: api/docs/docs.go (hunk 2 above)
Comment (AurevoirXavier) [reply api/docs/docs.go:245]: Could you update dj?

Comment (clearloop) [reply api/docs/docs.go:245]: Yep, only the old version will be broken.

**Comment by clearloop** [comment]
The ops of mmr_proof will be fixed in next pr

**Pull request merged**